provision/core/mariadb/provision.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+
+# MariaDB/MySQL
+#
+# Use debconf-set-selections to specify the default password for the root MariaDB
+# account. This runs on every provision, even if MariaDB has been installed. If
+# MariaDB is already installed, it will not affect anything.
+echo mariadb-server-10.3 mysql-server/root_password password "root" | debconf-set-selections
+echo mariadb-server-10.3 mysql-server/root_password_again password "root" | debconf-set-selections
+
+echo -e "\n * Setting up MySQL configuration file links..."
+
+if grep -q 'mysql' /etc/group; then
+echo " * mysql group exists"
+else
+echo " * creating mysql group"
+groupadd -g 9001 mysql
+fi
+
+if id -u mysql >/dev/null 2>&1; then
+echo " * mysql user present and has uid $(id -u mysql)"
+else
+echo " * adding the mysql user"
+useradd -u 9001 -g mysql -G vboxsf -r mysql
+fi
+
+mkdir -p "/etc/mysql/conf.d"
+echo " * Copying /srv/config/mysql-config/vvv-core.cnf to /etc/mysql/conf.d/vvv-core.cnf"
+cp -f "/srv/config/mysql-config/vvv-core.cnf" "/etc/mysql/conf.d/vvv-core.cnf"
+
+if ! vvv_apt_keys_has 'MariaDB'; then
+  # Apply the MariaDB signing keyg
+  echo " * Applying the MariaDB signing key..."
+  apt-key add /srv/config/apt-keys/mariadb.key
+fi
+
+if ! vvv_src_list_has "nginx.org"; then
+  cat <<VVVSRC >> /etc/apt/sources.list.d/vvv-sources.list
+# MariaDB 10.4 Amsterdam
```

Comment:
maybe also the same for this list in a file?

Reply:
A `vvv-sources.list` that gets copied into place sounds good

Reply:
maybe `mariadb-vvv-sources.list` so every tool can have their own file

Reply:
Maybe, but we need to be mindful that `vvv-sources.list` exists, otherwise `apt` will complain about the same source appearing more than once ( I currently get this for PHP locally at the moment )

Reply:
I could have the file, and do the same check i have right now, but instead of using HEREDOC using cat.

Reply:
A specific file would be better, force copying a `vvv-sources.list` into place avoids a lot of checks and potential bugs. We can always copy additional source files if we want to append extra sources

Reply:
Done